File: solutions/0150-evaluate-reverse-polish-notation/solution.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []

        for s in tokens:
            if (s in ['+', '-', '*', '/']):
                
                operand_02 = int(stack.pop())
                operand_01 = int(stack.pop())

                match s:
                    case '+':
                        res = operand_01 + operand_02
                    case '-':
                        res = operand_01 - operand_02
                    case '*':
                        res = operand_01 * operand_02
                    case '/':
                        res = operand_01 / operand_02

                stack.append(int(res))        

            else :
                stack.append(int(s))

        return stack.pop()

File: solutions/0150-evaluate-reverse-polish-notation/solution.ts
function evalRPN(tokens: string[]): number {
    let stack: number[] = []

    for (const t of tokens) {
        if (!['+', '-', '*', '/'].includes(t)) stack.push(parseInt(t))
        else {
            const op2 = stack.pop()
            const op1 = stack.pop()
            let result = 0

            switch (t) {
                case '+':
                    result = op1 + op2
                    break
                case '-':
                    result = op1 - op2
                    break
                case '*':
                    result = op1 * op2
                    break
                case '/':
                    result = Math.trunc(op1 / op2)
                    break
                default:
                    throw new Error('Unsupported operation.')
            }

            stack.push(result)
        }
    }

    return stack[0]
};
